mantis-control-plane/mantis-control-plane-core/src/main/java/io/mantisrx/server/master/resourcecluster/TaskExecutorRegistration.java

```diff
@@ -57,6 +67,28 @@ public class TaskExecutorRegistration {
   // TODO make this field non-null once no back-compat required.
   Map<String, String> taskExecutorAttributes;
 
+    @JsonCreator
+    public TaskExecutorRegistration(
+        @JsonProperty("taskExecutorID") TaskExecutorID taskExecutorID,
+        @JsonProperty("clusterID") ClusterID clusterID,
+        @JsonProperty("taskExecutorAddress") String taskExecutorAddress,
+        @JsonProperty("hostname") String hostname,
+        @JsonProperty("workerPorts") WorkerPorts workerPorts,
+        @JsonProperty("machineDefinition") MachineDefinition machineDefinition,
+        @JsonProperty("taskExecutorAttributes") Map<String, String> taskExecutorAttributes) {
+        this.taskExecutorID = taskExecutorID;
+        this.clusterID = clusterID;
+        this.taskExecutorAddress = taskExecutorAddress;
+        this.hostname = hostname;
+        this.workerPorts = workerPorts;
+        this.machineDefinition = machineDefinition;
+        this.taskExecutorAttributes = (taskExecutorAttributes == null) ? ImmutableMap.of() : taskExecutorAttributes;
+    }
+
+    public boolean containsAttributes(Map<String, String> requiredAttributes) {
+        return taskExecutorAttributes.entrySet().containsAll(requiredAttributes.entrySet());
+    }
```

Comment:
taskExecutorAttributes might be null on older version agents?

Reply:
no, when it gets deserialized, the null automatically gets converted into an empty map. I have tests covering that.